xray/services/summary.go

```diff
@@ -0,0 +1,116 @@
+package services
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/jfrog/jfrog-client-go/auth"
+	"github.com/jfrog/jfrog-client-go/http/jfroghttpclient"
+	"github.com/jfrog/jfrog-client-go/utils"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"net/http"
+)
+
+const (
+	summaryAPI = "api/v2/summary/"
+)
+
+func (ss *SummaryService) getSummeryUrl() string {
+	return ss.XrayDetails.GetUrl() + summaryAPI
+}
+
+// SummaryService returns the https client and Xray details
+type SummaryService struct {
+	client      *jfroghttpclient.JfrogHttpClient
+	XrayDetails auth.ServiceDetails
+}
+
+// NewSummaryService creates a new service to retrieve the version of Xray
+func NewSummaryService(client *jfroghttpclient.JfrogHttpClient) *SummaryService {
+	return &SummaryService{client: client}
+}
+
+// GetXrayDetails returns the Xray details
+func (ss *SummaryService) GetXrayDetails() auth.ServiceDetails {
+	return ss.XrayDetails
+}
+
+func (ss *SummaryService) GetBuildSummary(params XrayBuildParams) (*SummaryResponse, error) {
+	httpDetails := ss.XrayDetails.CreateHttpClientDetails()
+	url := fmt.Sprintf("%sbuild?build_name=%s&build_number=%s", ss.getSummeryUrl(), params.BuildName, params.BuildNumber)
+	if params.Project != "" {
+		url += "&" + projectKeyQueryParam + params.Project
+	}
+	resp, body, _, err := ss.client.SendGet(url, true, &httpDetails)
+	if err != nil {
+		return nil, err
+	}
+	if err = errorutils.CheckResponseStatus(resp, http.StatusOK); err != nil {
+		return nil, errorutils.CheckError(errorutils.GenerateResponseError(resp.Status, utils.IndentJson(body)))
+	}
+	var summaryResponse SummaryResponse
+	err = json.Unmarshal(body, &summaryResponse)
+	if err != nil {
+		return nil, errorutils.CheckError(err)
+	}
+	if summaryResponse.Errors != nil && len(summaryResponse.Errors) > 0 {
+		return nil, errorutils.CheckErrorf("Getting build-summery for build: %s failed with error: %s", summaryResponse.Errors[0].Identifier, summaryResponse.Errors[0].Error)
+	}
+	return &summaryResponse, nil
+}
+
+func (ss *SummaryService) GetArtifactSummary(artifactSummaryRequest ArtifactSummaryRequest) (*SummaryResponse, error) {
```

Comment:
I suggest we remove this API, becuase it's unclear when it should be used.
Let's also remove the structs this API uses.